frontend/src/features/properties/components/forms/subforms/EvaluationForm.tsx

```diff
@@ -52,45 +45,57 @@ const yearsArray = _.range(
 );
 const keyTypes = { ...EvaluationKeys, ...FiscalKeys };
 
-const findMatchingFinancial = (financials: IFinancial[], type: string, year?: number) => {
-  return financials?.find((financialsForYear: any) => {
-    const financial = financialsForYear[type.toLocaleLowerCase()];
+export const findMatchingFinancial = (financials: IFinancial[], type: string, year?: number) => {
+  return financials?.find((financialsForYear: IFinancial) => {
     return (
-      ((financial.date !== undefined && moment(financial.date).year() === year) ||
-        financial.fiscalYear === year) &&
-      financial.key === type
+      ((financialsForYear.date !== undefined && moment(financialsForYear.date).year() === year) ||
+        financialsForYear.fiscalYear === year ||
+        financialsForYear.year === year) &&
+      financialsForYear.key === type
     );
   });
 };
-const indexOfFinancial = (financials: IFinancial[], type: string, year?: number) =>
+export const indexOfFinancial = (financials: IFinancial[], type: string, year?: number) =>
   _.indexOf(financials, findMatchingFinancial(financials, type, year));
 /**
  * get a list of defaultEvaluations, generating one for NUMBER_OF_GENERATED_EVALUATIONS
  */
-export const defaultFinancials: any = yearsArray.map(year => {
-  return _.reduce(
-    Object.values(keyTypes) as string[],
-    (acc, type) => ({
-      ...acc,
-      [type.toLocaleLowerCase()]: {
-        date: type === EvaluationKeys.Assessed ? moment(year, 'YYYY').format('YYYY-MM-DD') : '',
-        year: year,
-        createdOn: '',
-        updatedOn: '',
-        fiscalYear: year,
-        key: type,
-        value: '',
-      },
-    }),
-    {} as IFinancialYear[],
-  );
-});
+export const defaultFinancials: any = _.flatten(
+  yearsArray.map(year => {
+    return _.reduce(
+      Object.values(keyTypes) as string[],
+      (acc, type) => [
+        ...acc,
+        {
+          date:
+            type === EvaluationKeys.Assessed || type === EvaluationKeys.Improvements
```

Comment:
dates are required for both improvements are assessed values, otherwise the db defaults to 1/1/0001